test/unit/webdriver/webdriver_test.py

```diff
@@ -298,6 +300,95 @@ def add_command(self):
         assert result == {}
         driver.delete_extensions()
 
+    @httpretty.activate
+    def test_create_session_with_custom_connection(self):
+        httpretty.register_uri(
+            httpretty.POST,
+            f'{SERVER_URL_BASE}/session',
+            body='{ "value": {"sessionId": "session-id", "capabilities": {"deviceName": "Android Emulator"}} }',
+        )
+
+        desired_caps = {
+            'deviceName': 'Android Emulator',
+            'app': 'path/to/app',
+        }
+
+        class CustomAppiumConnection(AppiumConnection):
+            # To explicitly check if the given executor is used
+            pass
+
+        init_args_for_pool_manager = {'retries': urllib3.util.retry.Retry(total=3, connect=3, read=False)}
+        custom_appium_connection = CustomAppiumConnection(
+            remote_server_addr=SERVER_URL_BASE, init_args_for_pool_manager=init_args_for_pool_manager
+        )
+
+        driver = webdriver.Remote(
+            custom_appium_connection, options=UiAutomator2Options().load_capabilities(desired_caps)
+        )
+
+        request = httpretty.HTTPretty.latest_requests[0]
+        assert request.headers['content-type'] == 'application/json;charset=UTF-8'
+        assert 'appium/python {} (selenium'.format(appium_version.version) in request.headers['user-agent']
+
+        request_json = json.loads(httpretty.HTTPretty.latest_requests[0].body.decode('utf-8'))
+        assert request_json.get('capabilities') is not None
+        assert request_json['capabilities']['alwaysMatch'] == {
+            'platformName': 'Android',
+            'appium:deviceName': 'Android Emulator',
+            'appium:app': 'path/to/app',
+            'appium:automationName': 'UIAutomator2',
+        }
+        assert request_json.get('desiredCapabilities') is None
+        assert driver.session_id == 'session-id'
+
+        assert isinstance(driver.command_executor, CustomAppiumConnection)
+
+    @httpretty.activate
+    def test_create_session_with_custom_connection_with_keepalive(self):
+        httpretty.register_uri(
+            httpretty.POST,
+            f'{SERVER_URL_BASE}/session',
+            body='{ "value": {"sessionId": "session-id", "capabilities": {"deviceName": "Android Emulator"}} }',
+        )
+
+        desired_caps = {
+            'deviceName': 'Android Emulator',
+            'app': 'path/to/app',
+        }
+
+        class CustomAppiumConnection(AppiumConnection):
+            # To explicitly check if the given executor is used
+            pass
+
+        init_args_for_pool_manager = {'retries': urllib3.util.retry.Retry(total=3, connect=3, read=False)}
+        custom_appium_connection = CustomAppiumConnection(
+            # keep alive has different route to set init args for the pool manager
+            keep_alive=True,
```

Comment:
Noticed the old method did not work when keep_alive was enabled, so modified to pass the arguments as the init